src/libs/highlevel/README.md

```diff
@@ -159,39 +181,43 @@ the most common use case for this callback is inside of functions that do not ta
 this function will be called, when any of the getter-functions is called on a non-existent key which is not part of any specification,
 and therefore has no specified default value.
 
+If you provide your own callback, it must interrupt the thread of execution in some way (e.g. by calling `exit()` or throwing an exception
+in C++). It *must not* return to the calling function.
+
 The handler will also be called whenever you pass `NULL` where a function expects an `ElektraError **`. In this case the error code will be
 `ELEKTRA_ERROR_CODE_NULL_ERROR`.
 
-The default callback simply logs the error with `ELEKTRA_LOG_DEBUG` and then calls `exit()` with the error code of the error.
-
-TODO: why DEBUG? It should be ERROR? And we should enable logging by default, otherwise users will not see the message?
-
-If you provide your own callback, it must interrupt the thread of execution in some way (e.g. by calling `exit()` or throwing an exception in C++). It
-*must not* return to the calling function.
+The default callback simply logs the error with `ELEKTRA_LOG_DEBUG` and then calls `exit()` with the error code of the error. It is expected
+that you implement your own callback, so that you get proper error message logged in your applications preferred format. Using the default
+callback is only viable for very simple applications, because you won't get any indication as to which key caused the error (unless you
+compiled Elektra with debug logging enabled).
 
 <a name="data-types"></a>
 ## Data Types
 
-The API supports the following types, which are taken from the CORBA specification:
-
-* **String**: a string of characters, represented by `KDB_TYPE_STRING` in metadata
+The API determines the data type of a given key, by reading its `type` metadata. The API supports the following types, 
+which are taken from the CORBA specification:
 
-TODO: please add better explanation, as started in PR #2377
-
-* **Boolean**: a boolean value `true` or `false`, represented by `KDB_TYPE_BOOLEAN` in metadata, in the KDB the raw value `"1"` is
+* **String**: a string of characters, represented by `string` in metadata
+* **Boolean**: a boolean value `true` or `false`, represented by `boolean` in metadata, in the KDB the raw value `"1"` is
```

Comment:
This is not according Elektra's boolean definition. See doc/decisions/bool.md and src/plugins/boolean/README.md